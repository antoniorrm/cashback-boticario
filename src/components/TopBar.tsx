import React from 'react';
import { AppBar, Toolbar, makeStyles, Button } from '@material-ui/core';

import LogoBoticatio from '../assets/img/logo-boticario-white.svg'

const useStyles = makeStyles(() => ({
	img: {
        width: "30vh",
        maxHeight: "64px",
		margin: 4,
	},
	toolbar: {
		display: 'flex',
		justifyContent: 'space-between',
	},
}))

const TopBar = () => {
  const classes = useStyles();

  return (
			<AppBar elevation={0}>
				<Toolbar className={classes.toolbar}>
					<img className={classes.img} src={LogoBoticatio} alt='Logo empresa' />
					<Button variant='outlined' color='inherit'>
						Sair
					</Button>
				</Toolbar>
			</AppBar>
		)
};

export default TopBar;
