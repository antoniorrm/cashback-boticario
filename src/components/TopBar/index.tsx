import React from 'react';
import { AppBar, Toolbar, makeStyles, Button } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import LogoBoticario from '../../assets/img/logo-boticario-white.svg'
import { signOut } from '../../store/modules/auth/actions';

const useStyles = makeStyles(() => ({
	img: {
        width: "30vh",
        maxHeight: "64px",
		margin: 4,
	},
	toolbar: {
		display: 'flex',
		justifyContent: 'space-between',
	},
}))

const TopBar = () => {
  const classes = useStyles();

  const dispatch = useDispatch()

  async function handleSignOut() {
      await dispatch(signOut())
  }

  return (
			<AppBar elevation={0}>
				<Toolbar className={classes.toolbar}>
					<img className={classes.img} src={LogoBoticario} alt='Logo empresa' />
					<Button variant='outlined' color='inherit' onClick={handleSignOut}>
						Sair
					</Button>
				</Toolbar>
			</AppBar>
		)
};

export default TopBar;
