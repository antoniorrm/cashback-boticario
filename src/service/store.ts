/* eslint-disable import/prefer-default-export */
import { store } from '../store'

export function getAccessToken() {
	return store.getState().auth.token
}