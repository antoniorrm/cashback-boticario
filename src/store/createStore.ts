import { createStore, applyMiddleware, Middleware, Reducer } from 'redux'
import { PersistState } from 'redux-persist'
import { AuthAction, AuthState } from './modules/auth/types'

export interface StoreState {
	_persist: PersistState
	auth: AuthState
}

export type StoreAction = AuthAction

export default (
	reducers: Reducer<StoreState, StoreAction>,
	middlewares: Middleware[]
) => {
	const enhancer = applyMiddleware(...middlewares)

	return createStore(reducers, enhancer)
}
