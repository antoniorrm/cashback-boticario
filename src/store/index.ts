import createSagaMiddleware from 'redux-saga'
import { AnyAction, Middleware, Store } from 'redux'
import { persistStore } from 'redux-persist'

import createStore from './createStore'
import rootReducer from './modules/rootReducer'
import rootSaga from './modules/rootSaga'
import persistReducers from './persistedReducers'

const sagaMiddleware = createSagaMiddleware()

const middlewares: Middleware[] = [sagaMiddleware]

const store = createStore(persistReducers(rootReducer), middlewares)
const persistor = persistStore(store as unknown as Store<any, AnyAction>)

sagaMiddleware.run(rootSaga)

export { store, persistor }
