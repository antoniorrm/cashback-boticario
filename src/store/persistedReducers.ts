import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { Reducer } from 'redux'
import { StoreAction, StoreState } from './createStore'

export default (reducers: Reducer<StoreState, StoreAction>) => {
	const persistedReducers = persistReducer(
		{
			key: 'cashback',
			storage,
			whitelist: ['auth'],
		},
		reducers
	)

	return persistedReducers
}
