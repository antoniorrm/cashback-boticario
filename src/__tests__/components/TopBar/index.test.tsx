/* eslint-disable no-undef */
import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import * as reactRedux from 'react-redux'

import TopBar from '../../../components/TopBar'
import { store } from '../../../store'

describe('TopBar component', () => {
	it('renders component correct', () => {
		const { container } = render(
			<Provider store={store}>
				<TopBar />
			</Provider>
		)
		expect(container).toMatchSnapshot()
	})
	it('renders image in component correct', () => {
		render(
			<Provider store={store}>
				<TopBar />
			</Provider>
		)
		const image = screen.getByAltText(/Logo empresa/i)
		expect(image).toHaveAttribute('src', 'logo-boticario-white.svg')
	})

	it('renders Button in component correct', () => {
		render(
			<Provider store={store}>
				<TopBar />
			</Provider>
		)
		const Button = screen.getByText(/Sair/i)
		expect(Button).toBeInTheDocument()
	})
	it('should be Button click in component', () => {
        const mockDispatch = jest.fn()
        const useDispatch = jest.spyOn(reactRedux, 'useDispatch')
        useDispatch.mockReturnValue(mockDispatch)
		render(
			<Provider store={store}>
				<TopBar />
			</Provider>
		)
		const Button = screen.getByRole('button', { name: 'Sair' })
		userEvent.click(Button)
		expect(mockDispatch).toHaveBeenCalled()
	})
})
