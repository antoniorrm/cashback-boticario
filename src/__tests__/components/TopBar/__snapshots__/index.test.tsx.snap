// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component renders component correct 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <img
        alt="Logo empresa"
        class="makeStyles-img-1"
        src="logo-boticario-white.svg"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Sair
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
